### PR TITLE
get_batch type and download binary

### LINKED ISSUE #49: add support for binary files in lims.get_file_content
add option to open files as binary in `lims.get_file_content`

### LINKED ISSUE #50: all_inputs documentat - list vs dict_values
The documentation for "Process.all_inputs" infers that the output is always a list type variable but the output variable type can be list or dict_values

Process.all_inputs(self, unique=True, resolve=False) produces a list type variable
Process.all_inputs(self, uniqued=True, resolve=True) produces a dict_values

 def all_inputs(self, unique=True, resolve=False):
        """Retrieving all input artifacts from input_output_maps.
        If unique is true, no duplicates are returned.

        :param unique: boolean specifying if the list of artifacts should be uniqued
        :param resolve: boolean specifying if the artifacts entities should be resolved through a batch query.
        :return: list of input artifacts._

### PR DESCRIPTION
**Pull request by tcezard**
Fix the return type of get_batch function: closes #50
Add option to download binary stream using lims.get_file_content: closes #49

Add some additional documentation to clarify create_batch limitation.

### CODE REVIEW

File: pyclarity_lims/lims.py [hunk 1]
```diff
@@ -637,7 +648,7 @@ def get_batch(self, instances, force=False):
             for node in root.getchildren():
                 instance = instance_map[node.attrib['limsid']]
                 instance.root = node
-        return instance_map.values()
+        return instances
```
Comment (lukbut) [inline pyclarity_lims/lims.py]: I do not see any changes made to "instances", are returning the object without modifying it? Are you relying on the variables retaining a pointer to the same data?

Comment (tcezard) [reply pyclarity_lims/lims.py]: Yes that is correct.
each instance that needs to be updated is passed on to a dict where the key is the instance id.
each instance's root is then updated which will update them within the original list.
It has the advantage of keeping the order of the original list the same.

Comment (lukbut) [reply pyclarity_lims/lims.py]: Ok, has this been tested? I'm a little wary of this functionality.

Comment (tcezard) [reply pyclarity_lims/lims.py]: yes in test_get_batch bellow

Comment (lukbut) [reply pyclarity_lims/lims.py]: Great.

**Pull request merged**